libcextract/FunctionDepsFinder.cpp

```diff
@@ -725,6 +697,58 @@ void FunctionDependencyFinder::Remove_Redundant_Decls(void) {
       }
     }
   }
+
+  // FIXME: use interval tree
+  ASTUnit::top_level_iterator it;
+  TypedefDecl *prev = nullptr;
+  for (it = AST->top_level_begin(); it != AST->top_level_end(); ++it) {
+    if (TypedefDecl *decl = dynamic_cast<TypedefDecl *>(*it)) {
```

Comment:
Please check if the Decl is in the closure. If not, then go to the next Decl.

Also avoid using dynamic_cast<>()